redbot/cogs/mod/events.py

```diff
@@ -68,6 +69,61 @@ async def check_mention_spam(self, message):
                         channel=None,
                     )
                     return True
+
+        if mention_spam["kick"]:
+            if len(mentions) >= mention_spam["kick"]:
+                try:
+                    await guild.kick(author, reason=_("Mention Spam (Autokick)"))
+                except discord.HTTPException:
+                    log.warning(
+                        "Failed to kick a member ({member}) for mention spam in server {guild}".format(
+                            member=author.id, guild=guild.id
+                        )
+                    )
+                else:
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        message.created_at,
+                        "kick",
+                        author,
+                        guild.me,
+                        _("Mention spam (Autokick)"),
+                        until=None,
+                        channel=None,
+                    )
+                    return True
+
+        if mention_spam["warn"]:
+            if len(mentions) >= mention_spam["warn"]:
+                try:
+                    await author.send(_("Please do not mass mention people!"))
+                except (discord.HTTPException, discord.Forbidden):
+                    try:
+                        await message.channel.send(
+                            _("{member}, Please do not mass mention people!").format(
+                                member=author.mention
+                            )
+                        )
```

Comment:
Even if this runs successfully, the case will not be created

Reply:
Hopefully solved.